Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/service/impl/ConceptMergeService.java

```diff
@@ -0,0 +1,224 @@
+        conceptsMergeBean.clear();
+        for (ConceptEntry entry : conceptEntries) {
+            if (entry.getWord() != null && !entry.getWord().isEmpty()) {
+                conceptsMergeBean.setWord(conceptsMergeBean.getWord() == null
+                        || conceptsMergeBean.getWord().length() < entry.getWord().length() ? entry.getWord()
+                                : conceptsMergeBean.getWord());
+                
+                String conceptIdWithName = entry.getWord().trim() + " -- " + entry.getId();
+                conceptsMergeBean.getConceptNamesWithIds().add(conceptIdWithName);
+            }
+
+            if (conceptsMergeBean.getSelectedPosValue() == null) {
+                conceptsMergeBean.setSelectedPosValue(entry.getPos().toLowerCase());
+            } else if (!conceptsMergeBean.getSelectedPosValue().equalsIgnoreCase(entry.getPos())) {
+                posWarning = true;
+            }
+
+            if (entry.getDescription() != null && !entry.getDescription().isEmpty()) {
+                conceptsMergeBean.getDescriptions().add(entry.getDescription().trim());
+            }
+
+            if (entry.getSynonymIds() != null && !entry.getSynonymIds().isEmpty()) {
+                conceptsMergeBean.getSynonymsids().addAll(Arrays.asList(entry.getSynonymIds().split(",")));
+            }
+
+            if (entry.getEqualTo() != null && !entry.getEqualTo().isEmpty()) {
+                conceptsMergeBean.getEqualsValues().add(entry.getEqualTo());
+            }
+            if (entry.getSimilarTo() != null && !entry.getSimilarTo().isEmpty()) {
+                conceptsMergeBean.getSimilarValues().add(entry.getSimilarTo());
+            }
+
+            if (entry.getTypeId() != null && conceptsMergeBean.getSelectedTypeId() == null) {
+                conceptsMergeBean.setSelectedTypeId(entry.getTypeId());
+            } else if (entry.getTypeId() != null) {
+                typeWarning = true;
+            }
+
+            if (entry.getWordnetId() != null && !entry.getWordnetId().isEmpty()) {
+                String[] wordNetIds = entry.getWordnetId().split(",");
+                conceptsMergeBean.getMergedIds()
+                        .addAll(Arrays.asList(wordNetIds).stream()
+                                .filter(wordNetId -> !wordNetId.trim().equalsIgnoreCase("")).map(wordNetId -> wordNetId)
+                                .collect(Collectors.toSet()));
+            }
+        }
+
+        if (posWarning) {
+            CPError errroMessage = new CPError("error_message_101",
+                    "There are difference in pos within the merging concepts.");
+            conceptsMergeBean.getErrorMessages().add(errroMessage);
+        }
+
+        if (typeWarning) {
+            CPError errroMessage = new CPError("error_message_102",
+                    "Merging concepts have different concept type.");
+            conceptsMergeBean.getErrorMessages().add(errroMessage);
+        }
```

Comment:
The warnings should show the different POSs/types used. I would get rid of the boolean flag, and add the warnings right where they are noticed (adding the POSs/types).